src/Microsoft.IdentityModel.Tokens/Encryption/EcdhKeyExchangeProvider.cs

```diff
@@ -56,51 +52,39 @@ public class EcdhKeyExchangeProvider //todo: rename to KeyExchangeProvider, or E
-        /// <param name="publicKey">The <see cref="ECDsaSecurityKey"/> that will be used for cryptographic operations and represents the public key.</param>
+        /// <param name="publicKey">The <see cref="SecurityKey"/> that will be used for cryptographic operations and represents the public key.</param>
         /// <param name="alg">alg header parameter value.</param>
         /// <param name="enc">enc header parameter value.</param>
         /// </summary>
-        public EcdhKeyExchangeProvider(ECDsaSecurityKey privateKey, ECDsaSecurityKey publicKey, string alg, string enc)
+        public EcdhKeyExchangeProvider(ECDsaSecurityKey privateKey, SecurityKey publicKey, string alg, string enc)
         {
             if (privateKey == null)
                 throw LogHelper.LogArgumentNullException(nameof(privateKey));
-            if (publicKey is null)
-                throw LogHelper.LogArgumentNullException(nameof(publicKey));
-
-            ValidateAlgAndEnc(alg, enc);
-
-            SetKeyDataLenAndEncryptionAlgorithm(alg, enc);
-            _ecParamsPrivate = privateKey.ECDsa.ExportParameters(true);
-            _ecParamsPublic = publicKey.ECDsa.ExportParameters(false);
-
-            ValidateCurves(nameof(privateKey), nameof(publicKey));
 
-            Initialize();
-        }
-
-        /// <summary>
-        /// Initializes a new instance of <see cref="EcdhKeyExchangeProvider"/> used for CEKs
-        /// <param name="privateKey">The <see cref="ECDsaSecurityKey"/> that will be used for cryptographic operations and represents the private key.</param>
-        /// <param name="publicKey">The <see cref="JsonWebKey"/> that will be used for cryptographic operations and represents the public key.</param>
-        /// <param name="alg">alg header parameter value.</param>
-        /// <param name="enc">enc header parameter value.</param>
-        /// </summary>
-        public EcdhKeyExchangeProvider(ECDsaSecurityKey privateKey, JsonWebKey publicKey, string alg, string enc)
-        {
-            if (privateKey == null)
-                throw LogHelper.LogArgumentNullException(nameof(privateKey));
             if (publicKey is null)
                 throw LogHelper.LogArgumentNullException(nameof(publicKey));
 
-            ValidateAlgAndEnc(alg, enc);
+            if (publicKey is ECDsaSecurityKey ecdsaKey)
+            {
+                _ecParamsPublic = ecdsaKey.ECDsa.ExportParameters(false);
+            }
+            else if (publicKey is JsonWebKey jwk)
+            {
+                _ecParamsPublic = GetEcParamsFromJwk(jwk);
+            }
+            else
+            {
+                throw LogHelper.LogArgumentException<ArgumentException>(
+                    nameof(publicKey),
+                    LogHelper.FormatInvariant(LogMessages.IDX11001, LogHelper.MarkAsNonPII(nameof(publicKey))));
```

Comment:
change error message, see comment in LogMessages.cs.

Reply:
Changed to IDX11002